src/benchq/resource_estimation/default_pipelines.py

```diff
@@ -233,74 +232,55 @@ def automatic_resource_estimator(
 ) -> ResourceInfo:
     """Pick the appropriate resource estimator based on the size of the program.
 
-    Currently chooses between GraphResourceEstimator and
-    ExtrapolationResourceEstimator and whether or not to use delayed gate synthesis
+    Currently, chooses between GraphResourceEstimator and
+    ExtrapolationResourceEstimator and whether to use delayed gate synthesis
     in the following order:
         1. GraphResourceEstimator without delayed gate synthesis
-        2. GraphResourceEstimator with delayed gate synthesis
-        3. ExtrapolationResourceEstimator without delayed gate synthesis
+        2. ExtrapolationResourceEstimator without delayed gate synthesis
+        3. GraphResourceEstimator with delayed gate synthesis
         4. ExtrapolationResourceEstimator with delayed gate synthesis
 
 
     Args:
-        program (QuantumProgram): program to estimate resources for
-        error_budget (ErrorBudget): budget for the error the program can tolerate
-        delayed_gate_synthesis (bool, optional): whether or not to decompose.
-            The gates into clifford + T before creating graph. Defaults to False.
-
-    Raises:
-        ValueError: if the problem is too big to estimate resources for
+        algorithm_implementation (AlgorithmImplementation): The algorithm to estimate
+            resources for.
+        hardware_model (BasicArchitectureModel): The hardware model to estimate
+            resources for.
+        decoder_model (Optional[DecoderModel], optional): The decoder model to
+            run the algorithm with. Defaults to None and returns no estimate in
+            that case.
 
     Returns:
-        GraphResourceEstimator: an appropriate resource estimator for the problem
+        ResourceInfo: The resources required to run the algorithm.
     """
-    pipeline: Any = None
     assert isinstance(algorithm_implementation.program, QuantumProgram)
 
-    prev_steps = algorithm_implementation.program.steps
+    graph_size = estimate_full_graph_size(algorithm_implementation, False)
+    reduced_graph_size = estimate_full_graph_size(algorithm_implementation, True)
```

Comment:
question: It seems like you're just getting graph sizes for the full graph and using that to determine which estimator to use. But in order to determine the size of the graph for the extrapolated case you need to get the graph sizes when the number of steps is reduced as well.

Reply:
This is why I had two separate if statements rather than a single if statement if you look at line 279 in the previous version I'm setting the step sizes back to the total number of steps.

Reply:
I missed the part where the number of steps did change the estimated size of a graph - fixed that